File: python/p001_multiples.py
"""
Code by Gonzalo Rolon Morinelli, Sep 8 2015.

The following code was originally written as a solution for Project Euler's
Problem# 1
For more information, goto: https://projecteuler.net/problem=1
"""


def sum_multiples(multipliers, limit):
    """Return the sum of all multiples of [multipliers] < the input limit."""
    multiples_sum = 0
    for idx in range(limit):
        for multiplier in multipliers:
            if idx % multiplier == 0:
                multiples_sum += idx
                break
    return multiples_sum

if __name__ == '__main__':
    # Problem solution
    print sum_multiples([3, 5], 1000)

File: js/p001_multiples.js
/**
Code by Gonzalo Rolon Morinelli, Dec 11 2015.

The following code was originally written as a solution for Project Euler's
Problem# 1
For more information, goto: https://projecteuler.net/problem=1
*/

/*jshint esnext: true*/

function sumMultiples (multipliers, limit) {
    // Return the sum of all multiples of [multipliers] < the input limit.
    var multiplesSum = 0;
    for (i = 0; i < limit; i++) {
        for (var multiple of multipliers) {
            if (i % multiple === 0) {
                multiplesSum += i;
                break;
            }
        }
    }
    return multiplesSum;
}

module.exports.sumMultiples = sumMultiples;

if (require.main === module) {
    // Problem solution
    console.log(sumMultiples([3, 5], 1000));
}
